homeassistant/components/enphase_envoy/diagnostics.py

```diff
@@ -37,11 +45,78 @@ async def async_get_config_entry_diagnostics(
     """Return diagnostics for a config entry."""
     coordinator: EnphaseUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    return async_redact_data(
-        {
-            "entry": entry.as_dict(),
-            "envoy_firmware": coordinator.envoy.firmware,
-            "data": coordinator.data,
-        },
-        TO_REDACT,
+    if TYPE_CHECKING:
+        assert coordinator.envoy.data
+    envoy_data: EnvoyData = coordinator.envoy.data
+    envoy: Envoy = coordinator.envoy
```

Comment:
```suggestion
    envoy_data = coordinator.envoy.data
    envoy = coordinator.envoy
```

`EnphaseUpdateCoordinator` is already typed so no need to add hints here